File: mps-n/src/mps-can/Signal.cpp
/*
 * Copyright (C) 2015 by Laurentiu-Gheorghe Crisan
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * As a special exception, you may use this file as part of a free
 * software library without restriction. Specifically, if other files
 * instantiate templates or use macros or inline functions from this
 * file, or you compile this file and link it with other files to
 * produce an executable, this file does not by itself cause the
 * resulting executable to be covered by the GNU General Public
 * License. This exception does not however invalidate any other
 * reasons why the executable file might be covered by the GNU Library
 * General Public License.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 */
#include "Signal.h"

namespace mps {
namespace can {

 Signal::Signal(Pt::uint32_t id)
: core::Signal(id)
, _byteCount(0)
, _byteOrder(0)
, _signalType(0)
, _modeValue(0)
, _pivotBit(0)
, _bitCount(0)
, _canDataType(0)
, _modeBitCount(0)
, _modeByteOrder(0)
, _modeFactor(0)
, _modeOffset(0)
, _modeCanDataType(0)
, _modePivotBit(0)
{
    registerProperty("id", *this, &Signal::msgId, &Signal::setMsgId);
    registerProperty("byteCount", *this, &Signal::byteCount, &Signal::setByteCount);
    registerProperty("byteOrder", *this, &Signal::byteOrder, &Signal::setByteOrder);
    registerProperty("signalType", *this, &Signal::signalType, &Signal::setSignalType);
    registerProperty("modeValue", *this, &Signal::modeValue, &Signal::setModeValue);
    registerProperty("pivotBit", *this, &Signal::pivotBit, &Signal::setPivotBit);
    registerProperty("bitCount", *this, &Signal::bitCount, &Signal::setBitCount);
    registerProperty("canDataType", *this, &Signal::canDataType, &Signal::setCanDataType);
    registerProperty("modeBitCount", *this, &Signal::modeBitCount, &Signal::setModeBitCount);
    registerProperty("modeByteOrder", *this, &Signal::modeByteOrder, &Signal::setModeByteOrder);
    registerProperty("modeFactor", *this, &Signal::modeFactor, &Signal::setModeFactor);
    registerProperty("modeOffset", *this, &Signal::modeOffset, &Signal::setModeOffset);
    registerProperty("modeDataType", *this, &Signal::modeCanDataType, &Signal::setModeCanDataType);
    registerProperty("modePivotBit", *this, &Signal::modePivotBit, &Signal::setModePivotBit);
}


Signal::~Signal(void)
{
}

}}

File: mps-m/Mp.Visual.WaveChart/Signal.cs

//    MeaProcess - Meaurement and Automation framework.
//    Copyright (C) 2015  Laurentiu-Gheorghe Crisan
//
//    This program is free software: you can redistribute it and/or modify
//    it under the terms of the GNU General Public License as published by
//    the Free Software Foundation, either version 3 of the License, or
//    (at your option) any later version.
//
//    This program is distributed in the hope that it will be useful,
//    but WITHOUT ANY WARRANTY; without even the implied warranty of
//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//    GNU General Public License for more details.
//
//    You should have received a copy of the GNU General Public License
//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
using System;
using System.Drawing;
using System.Collections;
using System.Runtime.Serialization;

namespace Mp.Visual.WaveChart
{
    /// <summary>
	///  The chart signal representation
	/// </summary>
    [Serializable]
    public class Signal
	{
        /// <summary>
        /// Default constructor.
        /// </summary>
        public Signal()
        { }

        /// <summary>
        /// The signal minimum.
        /// </summary>
        public double InitialMinimum = 1;
        
        /// <summary>
        /// The signal maximum.
        /// </summary>
        public double InitialMaximum = -1;

        /// <summary>
        /// The signal samplerate in Hz.
        /// </summary>        
        public double Samplerate = 100;
        
        /// <summary>
        /// The visible flag.
        /// </summary>        
        public bool Visible = true;
        
        /// <summary>
        /// The signal name.
        /// </summary>        
        public string Name;
        
        /// <summary>
        /// The signal comment.
        /// </summary>        
        public string Comment;
        
        /// <summary>
        /// The signal unit.
        /// </summary>        
        public string Unit;
        
        /// <summary>
        /// The signal color.
        /// </summary>        
        public Color LineColor = Color.White;
        
        /// <summary>
        /// The signal data point color.
        /// </summary>                        
        public Color PointColor = Color.Yellow;
        
        /// <summary>
        /// The signal line width.
        /// </summary>
        public uint LineWidth = 1;
        
        /// <summary>
        /// The signal point size.
        /// </summary>
        public uint PointSize = 3;

        /// <summary>
        /// The point visible flag.
        /// </summary>
        public bool PointsVisible = false;
        
        /// <summary>
        /// The Signal Y axis division.
        /// </summary>
        public uint YAxisDivision = 5;
        

        public int Index = 0;

        /// <summary>
        /// The Signal Y axis precision.
        /// </summary>
        /// <remarks>
        /// 12.23  => precision = 2
        /// 12.234 => precision = 3
        /// </remarks>
        public uint YAxisPrecision = 0;
        public double Minimum = -10;
        public double Maximum = 10;

        public object InternalHandle;
	}
}
